packages/backend/src/asset/service.ts

```diff
@@ -111,21 +132,60 @@ async function updateAsset(
   }
 }
 
+// soft delete
+async function deleteAsset(
+  deps: ServiceDependencies,
+  { id, deletedAt }: DeleteOptions
+): Promise<Asset | AssetError> {
+  if (!deps.knex) {
+    throw new Error('Knex undefined')
+  }
+  try {
+    // return error in case there is a peer or wallet address using the asset
+    const peer = await Peer.query(deps.knex)
+      .withGraphJoined('asset')
+      .where('assetId', id)
+      .first()
+    if (peer) {
+      return AssetError.InuseAsset
+    }
+
+    const walletAddress = await WalletAddress.query(deps.knex)
+      .withGraphJoined('asset')
```

Comment:
```suggestion
```
since we just need to check the `assetId` column, we don't need to fetch the asset from the assets table. (same above)